src/firefly/js/charts/ui/HistogramPlotly.jsx

```diff
@@ -0,0 +1,304 @@
+        };
+    }
+
+    shouldComponentUpdate(nextProps, ns) {
+        const bUpdate = sCompare(nextProps, ns);
+
+        if (bUpdate && nextProps !== this.props) {
+            this.chartingInfo = this.regenData(nextProps);
+        }
+        return bUpdate;
+    }
+
+    /*
+     * Expecting an 2 dimensional array of numbers
+     * each row is an array of 3 values:
+     * [0] number of points in a bin,
+     * [1] minimum of a bin
+     * [2] maximum of a bin
+     */
+    validateData(data) {
+        if (!data) { return false; }
+        let valid = true;
+        try {
+            data.sort(function(row1, row2){
+                // [1] is minimum bin edge
+                return row1[1]-row2[1];
+            });
+            if (data) {
+                for (var i = 0; i < data.length; i++) {
+                    if (data[i].length !== 3) {
+                        logError(`Invalid histogram data in row ${i} [${data[i]}]`);
+                        valid = false;
+                    } else if (data[i][1]>data[i][2]) {
+                        logError(`Histogram data row ${i}: minimum is more than maximum. [${data[i]}]`);
+                        valid=false;
+                    } else if (data[i+1] && Math.abs(data[i][2]-data[i+1][1])>1000*Number.EPSILON &&
+                        data[i][2]>data[i+1][1]) {
+                        logError(`Histogram data row ${i}: bin range overlaps the following row. [${data[i]}]`);
+                        valid=false;
+                    } else if (data[i][0] < 0) {
+                        logError(`Histogram data row ${i} count is less than zero, ${data[i][0]}`);
+                        valid=false;
+                    }
+                }
+                if (this.props.logs && this.props.logs.includes('x') && data[0][1]<=0) {
+                    logError('Unable to plot histogram: zero or subzero values on logarithmic scale');
+                    valid = false;
+                }
+            }
+        } catch (e) {
+            logError(`Invalid data passed to Histogram: ${e}`);
+            valid = false;
+        }
+        return valid;
+    }
+
+    componentWillMount() {
+        this.chartingInfo = this.regenData(this.props);
+    }
+
+    /*
+    afterRedraw(chart, pl) {
+        const {lastData}= this;
+
+        chart.on('plotly_hover', (eventData) => {
+            const pointNumber= eventData.points[0].pointNumber;
+            const str= get(lastData, ['text', `${pointNumber}`]);
+
+            this.setState( {
+                dataUpdate: { text : str }
+            } );
+        });
+    }
+    */
+
+
+    render() {
+        this.error = undefined;
+        const {dataUpdate} = this.state;
+        const {plotlyData, plotlyDivStyle, plotlyLayout}= this.chartingInfo;
+        const {width, height} = this.props;
+
+        return (
+            <div style={{width: width? Number(width): undefined,
+                         height: height? Number(height): undefined}}>
```

Comment:
Is there a reason why you are limiting width/height to numeric values only?  They cannot use '100px' or 'calc(100% - 20px)'?